x-pack/plugins/security_solution/public/detections/components/callouts/use_callout_storage.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { difference, fromPairs, identity } from 'lodash/fp';
+import { useCallback, useMemo } from 'react';
+import { useEffectOnce, useMap } from 'react-use';
+import { useMessagesStorage } from '../../../common/containers/local_storage/use_messages_storage';
+import { CallOutMessage } from './callout_types';
+
+export interface CallOutStorage {
+  getVisibleMessageIds: () => string[];
+  isVisible: (message: CallOutMessage) => boolean;
+  dismiss: (message: CallOutMessage) => void;
+}
+
+export const useCallOutStorage = (
+  messages: CallOutMessage[],
+  namespace: string = 'common'
+): CallOutStorage => {
+  const visibilityStateInitial = useMemo(() => createInitialVisibilityState(messages), [messages]);
+  const [visibilityState, setVisibilityState] = useMap(visibilityStateInitial);
+
+  const { getMessages, addMessage } = useMessagesStorage();
+
+  const dismissedMessagesKey = getDismissedMessagesStorageKey(namespace);
+
+  const getVisibleMessageIds = useCallback(() => {
+    return Object.entries(visibilityState)
+      .filter(([messageId, isVisible]) => isVisible)
+      .map(([messageId, isVisible]) => messageId);
+  }, [visibilityState]);
```

Comment:
Here I implemented it with normal state and callback hooks, because at this point this is the approach I saw in the codebase.

Question: do we use `useReducer`, redux or maybe even rx or sagas to build app logic? I think I saw some redux usage somewhere within `security_colution`...
Do we have some conventions/agreements on how to build it?